booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,716 @@
+        if (arrivalEndDate.isBefore(arrivalStartDate)) {
+          invalidDates.add(arrivalEndDate.toString());
+        }
+      }
+      return invalidDates.stream()
+        .map("The expected arrival dates  '%s' are not valid"::formatted)
+        .collect(Collectors.toSet());
+    }
+  );
+
+  private static final Predicate<JsonNode> IS_ISO_EQUIPMENT_CONTAINER_REEFER = (uteNode) -> {
+    var isoEquipmentNode = uteNode.path("ISOEquipmentCode");
+    return isReeferContainerSizeTypeCode(isoEquipmentNode.asText(""));
+  };
+  private static final Predicate<JsonNode> HAS_ISO_EQUIPMENT_CODE = (reqEquipNode) -> {
+    var isoEquipmentNode = reqEquipNode.path("ISOEquipmentCode");
+    return isoEquipmentNode.isTextual();
+  };
+  private static final Predicate<JsonNode> IS_ACTIVE_REEFER_SETTINGS_REQUIRED = (reqEquipNode) -> {
+    var norNode = reqEquipNode.path("isNonOperatingReefer");
+    if (HAS_ISO_EQUIPMENT_CODE.test(reqEquipNode) && IS_ISO_EQUIPMENT_CONTAINER_REEFER.test(reqEquipNode)) {
+      return !norNode.isMissingNode() && !norNode.asBoolean(false);
+    }
+    return false;
+  };
+
+  private static final Consumer<MultiAttributeValidator> ALL_REQ_EQUIP = (mav) -> mav.submitAllMatching("requestedEquipments.*");
+  private static final JsonContentCheck NOR_PLUS_ISO_CODE_IMPLIES_ACTIVE_REEFER = JsonAttribute.allIndividualMatchesMustBeValid(
+    "All requested Equipments where 'isNonOperatingReefer' is 'false' must have 'activeReeferSettings'",
+    ALL_REQ_EQUIP,
+    JsonAttribute.ifMatchedThen(
+      IS_ACTIVE_REEFER_SETTINGS_REQUIRED,
+      JsonAttribute.path("activeReeferSettings", JsonAttribute.matchedMustBePresent())
+    )
+  );
+
+  private static final JsonContentCheck ISO_EQUIPMENT_CODE_AND_NOR_CHECK = JsonAttribute.allIndividualMatchesMustBeValid(
+    "All requested Equipments where ISOEquipmentCode is reefer code must have 'isNonOperatingReefer' flag",
+    ALL_REQ_EQUIP,
+    JsonAttribute.ifMatchedThen(
+      IS_ISO_EQUIPMENT_CONTAINER_REEFER,
+      JsonAttribute.path("isNonOperatingReefer", JsonAttribute.matchedMustBePresent())
+    )
+  );
+
+  private static final JsonContentCheck IS_SHIPPER_OWNED_CONTAINER = JsonAttribute.customValidator(
+    "valid Requested Equipments",
+    (body) -> {
+      var issues = new LinkedHashSet<String>();
+      var requestedEquipments = body.path("requestedEquipments");
+      for (var rq: requestedEquipments) {
+        var isShipperOwned = rq.get("isShipperOwned").asBoolean(false);
+        var commodities = requestedEquipments.get("commodities");
+        if (isShipperOwned) {
+          for (var commodity : commodities) {;
+            var cargoGrossVolume = commodity.get("cargoGrossVolume");
+              var cargoGrossVolumeUnit = commodity.get("cargoGrossVolumeUnit");
+            if(cargoGrossVolume != null && (cargoGrossVolumeUnit  == null || cargoGrossVolumeUnit.isEmpty()) ) {
+              issues.add("the cargo gross volume unit must be provided");
+            }
```

Comment:
Nit / For next PR: Spacing is awkward here.

Reply:
done